src/SWP/Bundle/ContentBundle/Loader/ArticleLoader.php

```diff
@@ -125,7 +126,34 @@ public function load($type, $parameters, $responseType = LoaderInterface::SINGLE
                 ));
 
                 if ($route) {
-                    $articles = $this->dm->getReferrers($route, null, null, null, 'SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Article');
+                    $node = $this->dm->getNodeForDocument($route);
+                    $identifier = $node->getIdentifier();
+
+                    $queryStr = 'SELECT * FROM [nt:unstructured] as S ';
+                    $queryStr .= "WHERE S.phpcr:class='SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Article' ";
+                    $queryStr .= "AND S.route=$identifier";
```

Comment:
The query class has no setParameter method. The setParameter method is in the queryBuilder class. I could the query builder to build the query, but then I'd have to work out how to generate the query which is already there using the query builder, which seems like a waste of time. Seeing as the value of the parameter is provided by us, and not by an end user, I think we can safely put it directly in the query without using a placeholder. If you disagree, let me know, and I'll work out how to use the PhpcrQueryBuilder, but this will probably require trial and error because I can't find documentation on this query builder, only on the one returned by $this->dm->createQueryBuilder() which doesn't enable this kind of a query.